cEP-0020.md

```diff
@@ -0,0 +1,284 @@
+already known that newcomers face many barriers while attempting to contribute
+to the open source community for the first time. Some barriers they face include
+orientation issues that can potentially demotivate newcomers from placing
+their first contribution. On the other hand, gamification is widely used to
+engage and motivate people to accomplish tasks and improve their performance.
+Therefore, the goal of this project is to use gamification to orient and
+motivate newcomers to overcome onboarding barriers to contribute and engage
+with coala and its community.
+
+## Process involved in the Gamification system
+
+As the goal of this project is to make use of metrics and gamification to
+motivate and support newcomers to overcome the orientation barriers and
+contribute to the community, to achieve this goal we have selected three
+game design elements to help newcomers: Levels, Points, and Badges. For
+each of these elements, we will have a set of rules to describe the
+operation of the gamified environment. I.e.:
+
+- **Levels** group different tasks, enabling users to unlock specific and
+  contextualized sets of actions. Levels will be applied to make newcomers
+  focus on a given step before moving ahead on the contribution process.
+  We will divide our newcomer process into three levels and each level
+  will contain three tasks:
+
+  - Level-I:
+    - Joining our gitter channel and the community
+    - Running CI tools on your fork
+    - Running coala-ci on a popular github repo
+  - Level-II:
+    - Getting assigned to a difficulty/newcomer issue
+    - Creating and merging PR for that issue
+    - Reviewing at least one difficulty/newcomer issue
+  - Level-III:
+    - Getting assigned to a difficulty/low issue
+    - Creating and merging a PR for that issue
+    - Reviewing at least one difficulty/low issue
+
+- **Points** will be the most applied element out of three game design elements.
+  Acquiring points means that the actions performed by the users are in
+  accordance with what was expected. Points can be earned by means of performing
+  actions to the different tasks under levels. If a user wants to go to next
+  level he will have to collect a certain number of points from doing the tasks
+  from the previous level. But in case of Level-II and Level-III if a user is not
+  able to acquire the required number of points after performing all the tasks,
+  he will have to review more PRs to collect the required points.
+
+- **Badges** enable newcomers to follow their performance and compare to other
+  newcomers. **After completing the level three successfully the newcomer will
+  be a developer**. And he will be awarded a badge based on their performance
+  in each level and the total number of points.
+
+These all are just basic ideas; the process and tasks in the levels can be
+changed according to the use case and implementaion process.
+
+A similar process can also be extended and applied to developers to achieve more
```

Comment:
This sentence needs to be adjusted to indicate that post-developer is out of scope for this project, but it is right to say that this project could be used in the future to build something similar for developers.